core/src/main/java/com/scalar/db/transaction/consensuscommit/ParallelExecutor.java

```diff
@@ -47,81 +51,168 @@ public ParallelExecutor(ConsensusCommitConfig config) {
     this.parallelExecutorService = parallelExecutorService;
   }
 
-  public void prepare(List<ParallelExecutorTask> tasks) throws ExecutionException {
+  public void prepare(List<ParallelExecutorTask> tasks, String transactionId)
+      throws ExecutionException {
     try {
-      executeTasks(tasks, config.isParallelPreparationEnabled(), false);
+      // when the parallel preparation disabled, we stop running the tasks when one of them fails.
```

Comment:
The description basically explains what it does, but I think it's obvious and not much meaningful. So, it would be great if we can say something like why we do this and what kind of bad consequences would happen if `stopOnError=true` in prepare in the parallel mode.

Something like this;

Enabling stopOnError in parallel preparation would cause the corresponding rollback to be executed earlier than the preparation of records, which could result in left-unrecovered records even in a normal case. Thus, we disable stopOnError when parallel preparation is enabled.

Reply:
Thank you for the feedback! Updated the comment in the latest commit. Please take a look again!